src/main/scala/com/lightbend/sbt/MultiReleaseJarPlugin.scala

```diff
@@ -0,0 +1,100 @@
+package com.lightbend.sbt
+
+import sbt._
+import sbt.Keys._
+import sbt.plugins.JvmPlugin
+import sbt.{ AutoPlugin, Def, PluginTrigger, Plugins }
+
+import scala.collection.immutable
+
+object MultiReleaseJarPlugin extends AutoPlugin {
+
+  object MultiReleaseJarKeys {
+    val MultiReleaseJar = config("MultiReleaseJar") extend Compile
+    
+    val java9Directory = settingKey[File]("Where the java9 sources are")
+    val scala9Directory = settingKey[File]("Where the scala9 sources are")
+    val compileJava9 = taskKey[Seq[File]]("Compile Java9 sources into META-INF")
+    val metaInfVersionsTargetDirectory = settingKey[File]("Where the java9 classes should be written to." +
+      "Usually this is: ./target/scala-2.12/classes/META-INF/versions/9")
+  }
+  import MultiReleaseJarKeys._
+  
+  val autoImport = MultiReleaseJarKeys 
+  
+  override def requires: Plugins = JvmPlugin
+  override def trigger: PluginTrigger = allRequirements
+
+  override def projectConfigurations = Seq(MultiReleaseJar)
+  
+  // ---------------------------------------
+  // This could be taken from`java.util.jar.Attributes.MULTI_RELEASE`, but we don't want to requirerunning on JDK9! 
+  val MULTI_RELEASE_KEY = "Multi-Release"
+  // ---------------------------------------
+  
+  override def projectSettings = Seq(
+    
+//    // hijack source generators to compile the java9 sources
+//    sourceGenerators in Compile += Def.task { compileJava9.value }.taskValue,
+    
+    packageOptions in (Compile, packageBin) +=
+      Package.ManifestAttributes("Multi-Release" -> "true"),
```

Comment:
This enables on Java 9 the classloader to look for classes in the `META-INF/versions/9` directory.
It is ignored by JDK8